PrimePHP/PrimePHP.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/**
+ * PHP 8 Prime Sieve
+ * Class PrimeSieve
+ */
+
+class PrimeSieve
+{
+    private array $rawbits;
+
+    public static array $primeCounts = [
+        10 => 1,
+        100 => 25,
+        1000 => 168,
+        10000 => 1229,
+        100000 => 9592,
+        1000000 => 78498,
+        10000000 => 664579,
+        100000000 => 5761455,
+    ];
+
+    public function __construct(
+        private int $sieveSize = 1000000,
+    )
+    {
+        $rawbitSize = (int)($this->sieveSize + 1) / 2;
+        $this->rawbits = array_fill(0, $rawbitSize, true);
+    }
+
+    private function getBit(int $index): bool
```

Comment:
Type checking is done runtime, so despite being a fan of typed PHP for this you would get better performance by removing it and possibly using old-school PHPDoc instead.
```suggestion
    /**
     * @param int $index
     * @return bool
     */
    private function getBit($index)
```
Doing this to all of the code improves the performance by about 20%
Passes: 63, Time: 10.01975155800028, Avg: 0.15904367552381396, Limit: 1000000, Count: 78498, Valid: True
Passes: 75, Time: 10062ms, Avg: 134ms, Limit: 1000000, Count: 78498, Valid: True

Reply:
Also worth noting the python version comes in at 63 so this seams to be in the valid range of languages :P

Reply:
Hi, I do not see any real difference on my end.

## PHP (8.0.3)
### All types
```shell
Passes: 184, Time: 10015ms, Avg: 54ms, Limit: 1000000, Count: 78498, Valid: True
Passes: 187, Time: 10052ms, Avg: 53ms, Limit: 1000000, Count: 78498, Valid: True
Passes: 186, Time: 10017ms, Avg: 53ms, Limit: 1000000, Count: 78498, Valid: True
```

### No types (parameters / return)

```shell
Passes: 123, Time: 10027ms, Avg: 81ms, Limit: 1000000, Count: 78498, Valid: True
Passes: 191, Time: 10003ms, Avg: 52ms, Limit: 1000000, Count: 78498, Valid: True
Passes: 188, Time: 10004ms, Avg: 53ms, Limit: 1000000, Count: 78498, Valid: True
Passes: 183, Time: 10008ms, Avg: 54ms, Limit: 1000000, Count: 78498, Valid: True
```

### No types only getBit and clearBit parameters

```shell
Passes: 187, Time: 10012ms, Avg: 53ms, Limit: 1000000, Count: 78498, Valid: True
Passes: 184, Time: 10057ms, Avg: 54ms, Limit: 1000000, Count: 78498, Valid: True
Passes: 186, Time: 10031ms, Avg: 53ms, Limit: 1000000, Count: 78498, Valid: True
```

maybe this is where JIT comes in ? I have enabled it on my cli

```shell
php -i | grep -E '^opcache\.(enable_cli|jit|jit_buffer_size) '
opcache.enable_cli => On => On
opcache.jit => tracing => tracing
opcache.jit_buffer_size => 50M => 50M
```


And indeed now that it is getting a lot better thanks to the comments it is running quite a bit faster than Python on my computer

## Python (3.6)

```shell
Passes: 40, Time: 10.209030065016123, Avg: 0.25522575162540306, Limit: 1000000, Count: 78498, Valid: True
Passes: 40, Time: 10.075028802995803, Avg: 0.25187572007489506, Limit: 1000000, Count: 78498, Valid: True
Passes: 40, Time: 10.057532230013749, Avg: 0.2514383057503437, Limit: 1000000, Count: 78498, Valid: True
```

Reply:
I also removed casting when I removed the type, maybe that is what makes the difference?

Types
```
php8.0 PrimePHP.php
Passes: 170, Time: 10035ms, Avg: 59ms, Limit: 1000000, Count: 78498, Valid: True
```
No types
```
php8.0 PrimePHP.php
Passes: 186, Time: 10011ms, Avg: 53ms, Limit: 1000000, Count: 78498, Valid: True
```
No types + JIT
```
php8.0 -dopcache.enable_cli=1 -dopcache.jit_buffer_size=100M PrimePHP.php
Passes: 366, Time: 10003ms, Avg: 27ms, Limit: 1000000, Count: 78498, Valid: True
```

Reply:
No types, explicit casting:
`Passes: 173, Time: 10022ms, Avg: 57ms, Limit: 1000000, Count: 78498, Valid: True`
No types, implicit casting:
`Passes: 179, Time: 10002ms, Avg: 55ms, Limit: 1000000, Count: 78498, Valid: True`

Reply:
There's no real benefit on performance here by removing type of arguments and returns, so it's better to use it wherever you can. Result may vary so should run each benchmark few times.

I suggest to use `strict_types` so automatic type casting on arguments and returns will be avoided at all.
```
declare(strict_types=1);
```

Reply:
I added the strict_types and indeed I do not really see any performance difference on my end.